### PR TITLE
Add target option to `filter_min_distance`

### LINKED ISSUE #243: Add "minima" target to `filter_min_distance`
Current behaviour for `filter_min_distance` is to always select the feature with the largest threshold, even if the feature detection is targeting minima. This is an aspect of #230 that requires more urgent attention

### PR DESCRIPTION
**Pull request by w-k-jones**
Resolves #243, part of #230 

Changes should be covered by existing feature detection tests, however we should check that we are actually testing cases where features are located closer to each other than the distance threshold

Edit: this isn't covered by `test_filter_min_distance`, I'll look how this test works and add a target option

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Comment by w-k-jones** [comment]
Found the problem with test coverage, there are two tests called `test_filter_min_distance` in `test_feature_detection`! https://github.com/tobac-project/tobac/blob/ce390a36e40fe330e59a7c03ad6d02d77e54e812/tobac/tests/test_feature_detection.py#L63 and https://github.com/tobac-project/tobac/blob/ce390a36e40fe330e59a7c03ad6d02d77e54e812/tobac/tests/test_feature_detection.py#L198

@freemansw1 do you have any thoughts one which should be kept?

**Comment by w-k-jones** [comment]
On second consideration, the way that the original `test_filter_min_distance` works isn't fully deterministic, so it's impossible to see if `filter_min_distance` is doing the correct thing beyond removing one feature. I'm going to remove it and focus fully on the newer version of the test added by @freemansw1

**Comment by freemansw1** [comment]
> Found the problem with test coverage, there are two tests called `test_filter_min_distance` in `test_feature_detection`!
> 
> https://github.com/tobac-project/tobac/blob/ce390a36e40fe330e59a7c03ad6d02d77e54e812/tobac/tests/test_feature_detection.py#L63
> 
> and
> https://github.com/tobac-project/tobac/blob/ce390a36e40fe330e59a7c03ad6d02d77e54e812/tobac/tests/test_feature_detection.py#L198
> 
> @freemansw1 do you have any thoughts one which should be kept?

I believe that the second one is the more robust one and should be kept.

**Review by JuliaKukulies — APPROVED** [review]
Looks super @w-k-jones ! Just a small comment - otherwise approved from my side

File: tobac/tests/test_feature_detection.py [hunk 1]
```diff
@@ -248,6 +294,9 @@ def test_filter_min_distance(
     y_coord_name = "projection_coord_y"
     z_coord_name = "projection_coord_z"
 
+    if target not in ["maximum", "minimum"]:
```
[the remark this answers is not in the document]
Comment (w-k-jones) [reply tobac/tests/test_feature_detection.py:297]: Good plan, I'll make this change

[the remark this answers is not in the document]
Comment (JuliaKukulies) [reply tobac/tests/test_feature_detection.py]: Perfect, looks good now!

**Comment by w-k-jones** [comment]
> Just a small question: The case handling the equality of threshold values is actually identical for both `target=minimum` and `target=maximum`. Could this be moved outside the `if`/`elif` block such that it only appears once?
> 
> https://github.com/tobac-project/tobac/blob/4a311efa4fc41b223380bf4308af3b14be4be170/tobac/feature_detection.py#L1141-L1154
> 
> Otherwise this also looks good to me!

I also realised this when modifying the function to implement the KDTree approach in #249 . Can make the change here as well if you think it's worthwhile, although it may cause a merge conflict

**Review by freemansw1 — APPROVED** [review]
Really nice fix @w-k-jones. I'm happy with these changes.

**Comment by lettlini** [comment]
I don't think changing this is necessary @w-k-jones. It was just something I thought about when reviewing this PR.

**Review by lettlini — APPROVED** [review]

**Comment by w-k-jones** [comment]
> I don't think changing this is necessary @w-k-jones. It was just something I thought about when reviewing this PR.

It's a good suggestion, I changed it to be that way when re-implementing for v1.4 in #251 and it's a lot more elegant

Will merge this now and open #249

**Pull request merged**